lib/pwwka/receiver.rb

```diff
@@ -24,7 +24,7 @@ def self.subscribe(handler_klass, queue_name, routing_key: "#.#", block: true, p
         info "Receiving on #{queue_name}"
         receiver.topic_queue.subscribe(manual_ack: true, block: block) do |delivery_info, properties, payload|
           begin
-            payload = ActiveSupport::HashWithIndifferentAccess.new(JSON.parse(payload))
+            payload = parse_payload(payload)
             handler_klass.handle!(delivery_info, properties, payload)
             receiver.ack(delivery_info.delivery_tag)
             logf "Processed Message on %{queue_name} -> %{payload}, %{routing_key}", queue_name: queue_name, payload: payload, routing_key: delivery_info.routing_key
```

Comment:
We probably shouldn't log raw payloads.

Reply:
Agreed.  @igbanam I wonder if you should inline the contents of `parse_payload` here so you can modify the `logf` call to not show the payload in the log in the case we aren't parsing it?